account-decoder/src/parse_sysvar.rs

```diff
@@ -96,7 +97,24 @@ pub fn parse_sysvar(data: &[u8], pubkey: &Pubkey) -> Result<SysvarAccountType, P
                 .ok()
                 .map(SysvarAccountType::EpochRewards)
         } else {
-            None
+            // EpochRewards PartitionData accounts are owned by the sysvar
```

Comment:
is there an assumed owner check prior to this method being called? i don't see one asserted here and the method is public. i think there's potential for type confusion without.

---

i guess we could harden this by storing the target epoch and asserting that the pda matches the passed address

Reply:
There is an owner check here: https://github.com/solana-labs/solana/blob/1810feadc283b44655be33e7665941eaabd1522f/account-decoder/src/parse_account_data.rs#L98-L102

This fn and probably other stuff in `parse_sysvar.rs` could/should be demoted to pub(crate).

Reply:
I can do the demoting in a follow-up (after v1.18 branch is cut). We can also add an assert in each of the program-specific modules, if you like.